emulators/vpcs/Portfile

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                vpcs
+version             0.8c
+revision            126
+categories          emulators
+license             BSD
+maintainers         {gmail.com:lasticppc @lastic} openmaintainer
+platforms           darwin
+
+description         Virtual PC Simulator built from SVN.
+long_description    ${description} The VPCS can simulate up to 9 PCs. \
+                    You can ping/traceroute them, or ping/traceroute the other hosts/routers \
+                    from the virtual PCs when you study the Cisco routers in the Dynamips. \
+                    VPCS is not the traditional PC, it is just a program running on the Linux or Windows, \
+                    and only few network commands can be used in it. \
+                    But VPCS can give you a big hand when you study the Cisco devices in the Dynamips. \
+                    VPCS can replace the routers or VMware boxes which are used as PCs in the Dynamips network.\
+                    Try VPCS, it can save your CPU/Memory. It is very small.
+
+homepage            https://wiki.freecode.com.cn/doku.php?id=wiki:vpcs
+
+master_sites        https://sourceforge.net/code-snapshots/svn/v/vp/vpcs/code
+distfiles           vpcs-code-r126-trunk.zip
```

Comment:
Don't hardcode the `distfiles`. Instead set the `distname`. You want to create a variable for the `release` and add that here. The '.zip' ending is already handled by your use of `use_zip yes` below.

Reply:
If I don't specify the .zip , fetch tries without ?

Reply:
not if you use distname instead. Then the extension is automatically appended.

There are unfortunately lots of little things like that in MacPorts to learn -- that's why you're best to copy an existing working porfile if you can and make your small changes to it. It's harder to write them from scratch at the beginning.

Reply:
And another lesson learned, I was using ${distname} in distfiles but it actually replaces it,
got it now, .zip is indeed automatically added during fetch phase